### ISSUE TITLE
Request to adapt to fc2ppvdb.com.

### ISSUE THREAD

**Issue by maodun96** [KememChan/userscripts#1]
I like this script, but now msin.jp has shut down.
Currently, I am using fc2ppvdb.com.
Can you make the script compatible with this website?

**Comment by 68kdmd** [comment]
I came to ask this as well

**Comment by KememChan** [comment]
i can make the script for fc2ppvdb.com but i don't have the time to make that :(

**Comment by maodun96** [comment]
> i can make the script for fc2ppvdb.com but i don't have the time to make that :(

This is really regrettable. Your script has greatly improved my downloading experience, and I hope you have time in the future to update this script for fc2ppvdb. Thank you for your past scripts, and respect to you!

**Comment by KememChan** [comment]
@maodun96 @68kdmd  i just released the script for fc2ppvdb 991ca436070fccc6afce34fda6adcda18a60dd4b

**Comment by 68kdmd** [comment]
You’re the man. Thanks!

**Comment by 68kdmd** [comment]
It works perfectly. Awesome job!

**Comment by maodun96** [comment]
> @maodun96 @68kdmd i just released the script for fc2ppvdb [991ca43](https://github.com/KememChan/userscripts/commit/991ca436070fccc6afce34fda6adcda18a60dd4b)

The script runs perfectly; thank you for your development.
I have a few update requests and wonder if you could accommodate them.

1.Complete the film numbers: The film numbers for FC2PPVDB are missing the "FC2-" prefix. When using other plugins, it would be helpful to compare the full numbers with the local media library.
2.Dim the cover for films without download resources: I remember that you had a similar feature for the msin plugin, which is very useful.

Thank you again for your development; the plugin perfectly meets my needs. I appreciate your hard work.

**Comment by maodun96** [comment]
> > @maodun96 @68kdmd i just released the script for fc2ppvdb [991ca43](https://github.com/KememChan/userscripts/commit/991ca436070fccc6afce34fda6adcda18a60dd4b)
> 
> The script runs perfectly; thank you for your development. I have a few update requests and wonder if you could accommodate them.
> 
> 1.Complete the film numbers: The film numbers for FC2PPVDB are missing the "FC2-" prefix. When using other plugins, it would be helpful to compare the full numbers with the local media library. 2.Dim the cover for films without download resources: I remember that you had a similar feature for the msin plugin, which is very useful.
> 
> Thank you again for your development; the plugin perfectly meets my needs. I appreciate your hard work.

I use another script at https://github.com/qxinGitHub/searchAV. 
It should be very useful if you manage local videos with Jellyfin.

**Comment by maodun96** [comment]
> > > @maodun96 @68kdmd i just released the script for fc2ppvdb [991ca43](https://github.com/KememChan/userscripts/commit/991ca436070fccc6afce34fda6adcda18a60dd4b)
> > 
> > 
> > The script runs perfectly; thank you for your development. I have a few update requests and wonder if you could accommodate them.
> > 1.Complete the film numbers: The film numbers for FC2PPVDB are missing the "FC2-" prefix. When using other plugins, it would be helpful to compare the full numbers with the local media library. 2.Dim the cover for films without download resources: I remember that you had a similar feature for the msin plugin, which is very useful.
> > Thank you again for your development; the plugin perfectly meets my needs. I appreciate your hard work.
> 
> I use another script at https://github.com/qxinGitHub/searchAV. It should be very useful if you manage local videos with Jellyfin.

@KememChan 
Thank you for your update; everything is great! I didn't expect you to help me so much, and I really appreciate it!

I have one more small request: could you please add the prefix "FC2" to the ID on the details page? Currently, I am using a simple script to achieve this, but its effect is not very good.

```javascript
(function() {
    'use strict';

    function replaceText() {
        var textNodes = document.evaluate(
            '//text()[normalize-space()]',
            document,
            null,
            XPathResult.ORDERED_NODE_SNAPSHOT_TYPE,
            null
        );

        for (var i = 0; i < textNodes.snapshotLength; i++) {
            var node = textNodes.snapshotItem(i);
            node.textContent = node.textContent.replace(/ID：/g, 'ID：FC2-');
        }
    }

    window.addEventListener('load', replaceText);
})();
```
![image](https://github.com/user-attachments/assets/45bbc19b-9775-47a8-8dd4-72a178441579)
